tests/integration/framework/framework_test.go

```diff
@@ -17,28 +17,29 @@ package framework
 import (
 	"testing"
 
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
 	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
 
 	"istio.io/istio/pkg/test/framework"
-	"istio.io/istio/pkg/test/framework/components/environment"
-	"istio.io/istio/pkg/test/framework/components/istio"
+)
+
+var (
+	i   istio.Instance
+	env *kube.Environment
 )
 
 func TestMain(m *testing.M) {
 	framework.
 		NewSuite("framework_test", m).
 		Label(label.Presubmit).
-		RequireEnvironment(environment.Kube).
+		// The deployment must work. If you're breaking this, you'll break many integration tests.
+		Setup(istio.SetupOnKube(&i, nil)).
```

Comment:
This seems a little confusing ... `istio.SetupOnKube` would appear to be an environment-specific setup.  Shouldn't this be done in `EnvSetup`?

Reply:
Agreed. This is mostly historical. I added EnvSetup after I added SetupOnKube. I can do a separate PR to rationalize that?